pilot/pkg/model/push_context.go

```diff
@@ -386,6 +386,53 @@ func (rd ResourceDelta) IsEmpty() bool {
 	return len(rd.Subscribed) == 0 && len(rd.Unsubscribed) == 0
 }
 
+type ReasonStats map[TriggerReason]int
+
+func NewReasonStats(reasons ...TriggerReason) ReasonStats {
+	ret := make(ReasonStats)
```

Comment:
can we preallocate it to the number of TriggerReason we have? Not sure what Go's default size is for maps

Reply:
Considering that TriggerReason is not an iota, I haven't thought of an elegant way to maintain the "total number of TriggerReasons" temporarily. Given the small total number and the fact that map expansion is not as frequent as slice, memory optimization here should be negligible.